.github/workflows/publish_api_docs.yml

```diff
@@ -0,0 +1,64 @@
+name: Publish API Docs
+
+on:
+  # Manally run
+  workflow_dispatch:
+  # Pull request events for checking API docs
+  pull_request:
+  # Scheduled events for nightly API docs
+  schedule:
+    # UTC 00:00 everyday
+    - cron: "0 0 * * *"
+  # Events for API docs of new release
+  push:
+    branches:
+      - main
+    paths:
+      - VERSION
+  
+jobs: 
+  check_api_docs:
+    runs-on: ubuntu-latest
+    timeout-minutes: 15
+    container: asterinas/asterinas:0.15.2-20250613
+    if: github.event_name == 'pull_request'
+
+    steps: 
+      - uses: actions/checkout@v4
+
+      - name: Check API docs
+        run: ./tools/github_workflows/publish_api_docs.sh --dry-run
+
+  upload_nightly_api_docs:
```

Comment:
Follow the practice in `publish_osdk_and_ostd.yml` and downgrade `upload_nightly_api_docs` and `upload_release_api_docs` to steps. Otherwise, we will always get the following annoying "skipped checks" messages:

<img width="943" alt="image" src="https://github.com/user-attachments/assets/c5cdf633-2c32-4d6d-bf02-3fdfec70e377" />